src/hasura/migration-table-change.ts

```diff
@@ -0,0 +1,62 @@
+import {DangerDSLType, MarkdownString} from 'danger'
+import * as fs from 'fs'
+import * as R from 'ramda'
+
+/**
+ * Searches for Hasura migrations in edited files.
+ * If present, warns in case of:
+ * - rename of a field;
+ * - removal of a field;
+ * - "not null" restrictions on the existing field.
+ *
+ * @param danger Danger instance
+ * @param warn Danger warn function
+ * @param hasuraMigrationsPath paths to Hasura migrations
+ */
+export const migrationTableChange = (params: {
+  danger: DangerDSLType
+  hasuraMigrationsPath: string
+  warn: (message: MarkdownString, file?: string, line?: number) => void
+}) => {
+  const {danger, hasuraMigrationsPath, warn} = params
+  const hasuraMigrationsFiles = danger.git.fileMatch(`${hasuraMigrationsPath}/**/*`)
+
+  if (!hasuraMigrationsFiles.edited) {
+    return
+  }
+
+  R.forEach(
+    (filePath) =>
+      R.compose<[string], string, string[], string[], string[], void>(
+        R.forEach((statement) => {
+          const isAlterTableStatement = /^ALTER TABLE/.test(statement)
```

Comment:
@quesabe let's also account for [views](https://www.postgresql.org/docs/current/sql-alterview.html) and [materialised views](https://www.postgresql.org/docs/current/sql-altermaterializedview.html).